pkg/apis/core/validation/validation.go

```diff
@@ -2262,6 +2279,14 @@ func ValidatePersistentVolumeClaimUpdate(newPvc, oldPvc *core.PersistentVolumeCl
 
 	allErrs = append(allErrs, ValidateImmutableField(newPvc.Spec.VolumeMode, oldPvc.Spec.VolumeMode, field.NewPath("volumeMode"))...)
 
+	if !apiequality.Semantic.DeepEqual(oldPvc.Spec.VolumeAttributesClassName, newPvc.Spec.VolumeAttributesClassName) {
```

Comment:
@sunnylovestiramisu Does the pvc allow the volumeAttributesClass update from non-nil value to nil?

Reply:
I do not think this should be allowed, because if the underlying storage provider does not have a mechanism to set default mutable_parameters value in the volumeAttributesClass, passing in a nil volumeAttributesClass will trigger a modifyVolume failure. The validations suggested in the SPEC is only for mutable_parameters but not the volumeAttributesClassName.

@msau42 Do you agree with this?

Reply:
@sunnylovestiramisu 

Unfortunately, it's impossible to do that in [validation.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go). And there's no an example. If we try to do that, A possible way is to introduce a new admission plugin to validate it. I'm not sure if it's a good idea.

Why we need to validate the parameters field of the corresponding volumeattributesclass in the validation of pvc updation? Is it immutable field for a volumeattributesclass? If we need to validate the parameters field, we should validate it in the validation of volumeattributesclass creation, not in the validation of pvc updation and creation.

Reply:
ValidateVolumeAttributesClass validation funcs:

https://github.com/carlory/kubernetes/blob/kep-3751/pkg/apis/storage/validation/validation.go#L590

Reply:
I see you added: ```if oldPvc.Spec.VolumeAttributesClassName != nil && newPvc.Spec.VolumeAttributesClassName == nil {
				allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "volumeAttributesClassName"), "update from non-nil value to nil is forbidden"))
			}```

This should be sufficient?

Reply:
I think it's sufficient.